scripts/k8conduit.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+# Connect to a backend service via an app instance
+#
+
+# TODO
+#
+# - usually more than one redis service. Set default and how to override?
+# - test against azure redis service when available
+# - confirm before running if interactive, a flag to run without confirmation?
+#
+
+Help()
+{
+   echo
+   echo "Script to connect to a k8 backing service via an app service"
+   echo
+
+   echo "Syntax:"
+   echo "   k8conduit [-h|i] app-name -- command [args]"
+   echo "      Connect to the default database for app-name"
+   echo
+   echo "or k8conduit [-h|i] -d db-name -k key-vault app-name -- command [args]"
+   echo "      Connect to a specific database from app-name"
+   echo "      Requires a secret containing the DB URL in the specified Azure KV,"
+   echo "      with name {db-name}-database-url"
+   echo
+   echo "options:"
+   echo "   d            database name, required if connecting to a db other than the app default."
+   echo "   i            input file for a restore. Only valid for command psql"
+   echo "   k            key vault that holds the DB URL."
+   echo "   h            print this help."
+   echo
+   echo "parameters:"
+   echo "   app-name     app name to connect to."
+   echo "   command      command to run."
+   echo "                  valid commands are psql, pg_dump, redis-cli"
+   echo "   args         args for the command"
+}
+
+InitSetup()
+{
+   if [ "${RUNCMD}" != "psql" ] && [ "${RUNCMD}" != "pg_dump" ] && [ "${RUNCMD}" != "redis-cli" ]; then
+      echo "Error: invalid command ${RUNCMD}"
+      echo "Only valid options are psql, pg_dump or redis-cli"
+      exit
+   fi
+
+   # If an input file is given, check it exists and is readable
+   if [ "${Inputfile}" != "" ] && [ ! -r "${Inputfile}" ]; then
+      echo "Error: invalid input file"
+      exit
+   fi
+
+   # Set some defailt vars
+   TMOUT=28800 # 8 hour timeout
+   DB_PORT=5432
+   REDIS_PORT=6379
+}
+
+
+CheckInstance()
+{
+   if [ "$INSTANCE" = "" ]; then
+	   echo "Error: Must provide instance name as parameter e.g. apply-qa, apply-review-1234"
+	   exit
+   fi
+   # make sure it's LC
+   INSTANCE=$(echo "${INSTANCE}" | tr '[:upper:]' '[:lower:]')
+   # Lets check the container exists and we can connect to it first
+   if ! kubectl -n development exec -i deployment/"${INSTANCE}" -- echo; then
```

Comment:
Is this tied to the development namespace?

Reply:
Changed to get the namespace from the deployment